### ISSUE TITLE
Verify that the Swift ↔ C string conversion is correct

### ISSUE THREAD

**Issue by igor-makarov** [igor-makarov/SwiftPostal#1]
I've made the Swift ↔ C string conversion for my best knowledge. 

However, I'm not sure I've done it in a safe manner, and whether it leaks.

Since the code is primarily for server side, not leaking is extra important.

Perhaps someone with more Swift ↔ C bridging experise can weigh in?

@paulofaria ?
@Danappelxx ?

**Comment by paulofaria** [comment]
Did you profile the application and checked for leaks? You can do that from Xcode.

**Comment by igor-makarov** [comment]
At first, I tried profiling the XCTest but that just kept dropping the process.
I ended up building a ProfilerHelper console app that I profile. I've found and fixed on memory leak.

I have trouble understanding something. When I wrap my code iteration in an autorelease pool, everything gets released when the pool is drained.

What happens when there's no pool, like on Linux? Experimenting on a docker container with limited memory results in memory reaching the limit and kinda staying there. The program itself manages to run without problem. What's going on?
